decimal_test.go

```diff
@@ -31,6 +31,17 @@ func TestSetDefaultPrecision(t *testing.T) {
 	})
 }
 
+func TestPrecOutOfRange(t *testing.T) {
+	defer SetDefaultPrecision(maxPrec)
+
+	require.Equal(t, uint8(19), defaultPrec)
+
+	SetDefaultPrecision(10)
+
+	_, err := Parse("0.12345678901234569")
+	require.Equal(t, ErrPrecOutOfRange, err)
```

Comment:
```suggestion
	require.ErrorIs(t, err, ErrPrecOutOfRange)
```

You could enable testifylint in golangci-lint to detect this